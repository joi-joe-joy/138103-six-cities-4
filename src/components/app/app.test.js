import React from "react";
import renderer from "react-test-renderer";
import App from "./app.jsx";

const offers = [
  `Beautiful &amp; luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
  `Wood and stone place`
];

describe(`Render App`, () => {
  it(`Render with offers and rentCount`, () => {
    const tree = renderer
      .create(<App
        rentCount={3}
        offers={offers}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`App without offers and rentCount`, () => {
    const tree = renderer
      .create(<App
        offers={[``]}
        rentCount={0}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

