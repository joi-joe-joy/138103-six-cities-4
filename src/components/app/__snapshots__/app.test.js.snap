// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Render with offers 1`] = `
<main
  className="page__main page__main--index"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="#"
            onClick={[Function]}
          >
            <span>
              Paris
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Brussels
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          2
           places to stay in 
          Paris
        </b>
        <form
          action="#"
          className="places__sorting"
          method="get"
        >
          <span
            className="places__sorting-caption"
          >
            Sort by
          </span>
          <span
            className="places__sorting-type"
            tabIndex="0"
          >
            Popular
            <SvgrURL
              height="4"
              width="7"
            />
          </span>
          <ul
            className="places__options places__options--custom places__options--opened"
          >
            <li
              className="places__option places__option--active"
              tabIndex="0"
            >
              Popular
            </li>
            <li
              className="places__option"
              tabIndex="0"
            >
              Price: low to high
            </li>
            <li
              className="places__option"
              tabIndex="0"
            >
              Price: high to low
            </li>
            <li
              className="places__option"
              tabIndex="0"
            >
              Top rated first
            </li>
          </ul>
          <select
            className="places__sorting-type"
            defaultValue="popular"
            id="places-sorting"
          >
            <option
              className="places__option"
              value="popular"
            >
              Popular
            </option>
            <option
              className="places__option"
              value="to-high"
            >
              Price: low to high
            </option>
            <option
              className="places__option"
              value="to-low"
            >
              Price: high to low
            </option>
            <option
              className="places__option"
              value="top-rated"
            >
              Top rated first
            </option>
          </select>
        </form>
        <div
          className="places__list cities__places-list tabs__content"
        >
          <article
            className="place-card cities__place-card"
            onMouseOver={[Function]}
          >
            <div
              className="place-card__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              className="place-card__image-wrapper cities__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Canal View Prinsengracht"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-01.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    280
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  onClick={[Function]}
                  type="button"
                >
                  <SvgrURL
                    height="18"
                    width="17"
                  />
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  onClick={[Function]}
                >
                  Canal View Prinsengracht
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                House
              </p>
            </div>
          </article>
          <article
            className="place-card cities__place-card"
            onMouseOver={[Function]}
          >
            <div
              className="place-card__image-wrapper cities__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Nice, cozy, warm big bed apartment"
                  className="place-card__image"
                  height="200"
                  src="img/apartment-01.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    80
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  onClick={[Function]}
                  type="button"
                >
                  <SvgrURL
                    height="18"
                    width="17"
                  />
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  onClick={[Function]}
                >
                  Nice, cozy, warm big bed apartment
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                Private Room
              </p>
            </div>
          </article>
        </div>
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="map cities__map"
        >
          <div
            id="map"
            style={
              Object {
                "height": "100%",
              }
            }
          />
        </section>
      </div>
    </div>
  </div>
</main>
`;
